src/lib/gssapi/krb5/accept_sec_context.c

```diff
@@ -633,6 +638,81 @@ process_checksum(OM_uint32 *minor_status, krb5_context context,
     return status;
 }
 
+/*
+ * Determine the delegation type to use based on what we have and the policy.
+ */
+static krb5_error_code
+parse_delegation_policy(krb5_context context, char *policy,
+                        gss_cred_usage_t cred_usage, krb5_boolean have_tgt,
+                        delegation_types *out_delegation_type)
+{
+    char *token, *save;
+
+    *out_delegation_type = DELEG_NONE;
+
+    token = strtok_r(policy, ",", &save);
+    while (token != NULL) {
+        if (have_tgt && strcmp(token, "client-tgt") == 0) {
+            *out_delegation_type = DELEG_CLIENT_TGT;
+            break;;
+        }
+        if (strcmp(token, "proxy-creds") == 0 &&
+            (cred_usage == GSS_C_BOTH || cred_usage == GSS_C_INITIATE)) {
```

Comment:
Can cred_usage be GSS_C_INITIATE?  This is from the gss_accept_sec_context() acceptor_cred_handle, right?

Reply:
That's just a preparation to move parse_delegation_policy() to s4u_gss_glue.c  and to call it also from gss_acquire_cred_impersonate_name() in order to handle the cred_store option.